probcal/utils/experiment_utils.py

```diff
@@ -1,50 +1,121 @@
 import random
 from pathlib import Path
-from typing import Type, Optional
+from typing import Type, Optional, Union
 import os
 
 import lightning as L
 import numpy as np
 import torch
 from lightning.pytorch.callbacks import ModelCheckpoint
 
-from probcal.data_modules import TabularDataModule, COCOPeopleDataModule
+from probcal.data_modules import TabularDataModule, COCOPeopleDataModule, OodCocoPeopleDataModule
 from probcal.enums import DatasetType, ImageDatasetName, TextDatasetName
 from probcal.enums import HeadType
 from probcal.models import GaussianNN
 from probcal.models import NegBinomNN
 from probcal.models import PoissonNN
-from probcal.models.backbones import MLP
+from probcal.models.backbones import MLP, ViT, MNISTCNN
 from probcal.models.discrete_regression_nn import DiscreteRegressionNN
-from probcal.utils.configs import TrainingConfig
+from probcal.utils.configs import TrainingConfig, TestConfig
+from probcal.utils.generic_utils import partialclass
 
 GLOBAL_DATA_DIR = 'data'
 
 
-def get_model(config: TrainingConfig, return_initializer: bool = False) -> DiscreteRegressionNN:
+def get_model(config: Union[TrainingConfig, TestConfig], return_initializer: bool = False) -> DiscreteRegressionNN:
```

Comment:
This function might warrant future cleanup since there are a lot of edge cases we handle that probably don't apply here (MultiClassNN as a regressor is one example). For now, it's just a lot of extra complexity